nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/util/JSONPropertyWriter.java

```diff
@@ -49,39 +50,174 @@
  */
 public class JSONPropertyWriter {
 
-    /** Utility class. */
-    private JSONPropertyWriter() {
+    /**
+     * @since 9.1 the date time format
+     */
+    protected DateTimeFormat dateTimeFormat;
+
+    /**
+     * @since 9.1 the baseUrl that can be used to locate blob content
+     */
+    protected String filesBaseUrl;
+
+    /**
+     * @since 9.1 The prefix to append to field name
+     */
+    protected String prefix;
+
+    /**
+     * @since 9.1 whether or not this writer write null values
+     */
+    protected boolean writeNull = true;
+
+    /**
+     * @since 9.1 whether or not this writer write empty list or object
+     */
+    protected boolean writeEmpty = true;
+
+    /**
+     * Instantiate a JSONPropertyWriter.
+     */
+    protected JSONPropertyWriter() {
+        // Default constructor
     }
 
     /**
-     * Converts the value of the given core property to JSON. The given filesBaseUrl is the baseUrl that can be used to
-     * locate blob content and is useful to generate blob URLs.
+     * Copy constructor.
+     *
+     * @since 9.1
      */
-    public static void writePropertyValue(JsonGenerator jg, Property prop, DateTimeFormat dateTimeFormat,
-            String filesBaseUrl) throws PropertyException, JsonGenerationException, IOException {
+    protected JSONPropertyWriter(JSONPropertyWriter writer) {
+        this.dateTimeFormat = writer.dateTimeFormat;
+        this.filesBaseUrl = writer.filesBaseUrl;
+        this.prefix = writer.prefix;
+        this.writeNull = writer.writeNull;
+    }
+
+    /**
+     * @return a {@link JSONPropertyWriter} instance with {@link DateTimeFormat#W3C} as date time formatter.
+     * @since 9.1
+     */
+    public static JSONPropertyWriter create() {
+        JSONPropertyWriter writer = new JSONPropertyWriter();
+        writer.dateTimeFormat = DateTimeFormat.W3C;
+        return writer;
+    }
+
+    /**
+     * @return a new {@link JSONPropertyWriter} filled with the previous configuration and the input dateTimeFormat.
+     * @since 9.1
+     */
+    public JSONPropertyWriter dateTimeFormat(DateTimeFormat dateTimeFormat) {
+        JSONPropertyWriter writer = new JSONPropertyWriter(this);
```

Comment:
Do you really need to return a new one each time? Fluent style doesn't necessarily imply that, and it would be faster without. I'd modify the existing one in place.